src/tracing/esp32_trace/include/matter/tracing/macros_impl.h

```diff
@@ -21,23 +21,47 @@
 #error "Tracing macros seem to be double defined"
 #endif
 
+#include <lib/core/CHIPError.h>
+
 namespace Insights {
-class ESP32Backend
+
+/*  The ESP32Filter namespace contains two apis AddStringToBlacklist and
+ *  RemoveStringFromBlacklist to provide some flexibility to the user to add or
+ *  remove string from the list of blacklisted strings which the user wants to
+ *  filter out.
+ */
+namespace ESP32Filter {
+
+CHIP_ERROR AddStringToBlacklist(const char * aLabel);
+void RemoveStringFromBlacklist(const char * aLabel);
+} // namespace ESP32Filter
+
+/* The ESP32Trace class helps to send the entry and exit traces of the scope
+ * macros in the code along with some trace info like largest free block,
+ * min free heap,start free heap ,etc.
+ * For ex:
+ * MTR_TRC: Entry - HandleSigma1_and_SendSigma2 - CASESession Min Free heap - 35640 - LFB - 22528 Start free heap - 44124
+ * MTR_TRC: Exit - HandleSigma1_and_SendSigma2 - CASESession Min Free heap - 35640 - LFB - 22528 Start free heap - 44216
+ * It does this generically for all the scoped trace macros in the code for better debugging if required and
+ * reports these traces to esp-insights cloud using ESP_DIAG_LOG.
+ */
+class ESP32Trace
```

Comment:
Named should probably contain a `Scope` suffix since it seems to behave as such.